base/number.jl

```diff
@@ -3,6 +3,15 @@
 ## generic operations on numbers ##
 
 isinteger(x::Integer) = true
+iszero(x::Number) = x == 0 # faster than x == zero(x) for complex, bigfloat, etc.
+
+"""
+    iszero(x)
+
+Return `true` if `x == zero(x)`; if `x` is an array, this checks whether
+all of the elements of `x` are zero.
+"""
+iszero(x) = x == zero(x) # fallback method
```

Comment:
`iszero("cat")` seems questionable to me... we don't define `isinteger("cat")` either.